prov/sockets/src/sock_fabric.c

```diff
@@ -319,7 +320,10 @@ static int sock_fabric(struct fi_fabric_attr *attr,
 	fab->fab_fid.ops = &sock_fab_ops;
 	*fabric = &fab->fab_fid;
 	atomic_init(&fab->ref, 0);
-	sock_fab_add_to_list(fab);
+#if ENABLE_DEBUG	
+	atomic_init(&fab->num_send_msg, 0);
+#endif
```

Comment:
You could simplify this by having a function, sock_drop_packet() which returns true if it should drop a packet.  In non-debug builds, that function would always return 0, which the compiler should optimize out.